scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -885,12 +885,23 @@ protected void calculateKeys(SecretKey masterSecret) {
 	 */
 	private SecretKey generateMasterSecret(SecretKey premasterSecret) {
 		byte[] randomSeed = Bytes.concatenate(clientRandom, serverRandom);
-		byte[] secret = PseudoRandomFunction.doPRF(session.getCipherSuite().getThreadLocalPseudoRandomFunctionMac(),
-				premasterSecret, Label.MASTER_SECRET_LABEL, randomSeed);
+		byte[] secret = deriveMasterSecret(premasterSecret, randomSeed);
 		SecretKey masterSecret = SecretUtil.create(secret, "MAC");
 		Bytes.clear(secret);
 		return masterSecret;
 	}
+	
+	/**
+	 * Derives the master secret from the premaster secret and the random seed.
+	 * 
+	 * @param premasterSecret the Pre Master Secret Key
+	 * @param randomSeed the random seed bytes
+	 * @return the derived master secret bytes
+	 */
+	protected byte[] deriveMasterSecret(SecretKey premasterSecret, byte[] randomSeed) {
```

Comment:
Did you have a look at the [SecretUtil](https://github.com/eclipse/californium/blob/master/scandium-core/src/main/java/org/eclipse/californium/scandium/util/SecretUtil.java)? Maybe you can adapt the return type to a `SecretKey`?